File: gles-lib/src/main/cpp/egl/OffscreenSurface.cpp
//
// Created by gtbluesky on 2019-07-16.
//

#include "OffscreenSurface.h"

OffscreenSurface::OffscreenSurface(EglCore *eglCore, int width, int height)
        : EglSurfaceBase(eglCore) {
    createOffscreenSurface(width, height);
}

void OffscreenSurface::release() {
    releaseEglSurface();
}

File: gles-lib/src/main/java/com/gtbluesky/gles/egl/OffscreenSurface.kt
package com.gtbluesky.gles.egl

/**
 * Off-screen EGL surface (pbuffer).
 * It's good practice to explicitly release() the surface, preferably from a "finally" block.
 */
class OffscreenSurface(eglCore: EglCore, width: Int, height: Int) : EglSurfaceBase(eglCore) {

    init {
        /**
         * Creates an off-screen surface with the specified width and height.
         */
        createOffscreenSurface(width, height)
    }

    /**
     * Releases any resources associated with the surface.
     */
    fun release() {
        releaseEglSurface()
    }
}